src/dotnet/APIView/APIViewWeb/LeanControllers/APIRevisionsController.cs

```diff
@@ -0,0 +1,115 @@
+using APIViewWeb.Helpers;
+using APIViewWeb.LeanModels;
+using APIViewWeb.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using APIViewWeb.Managers.Interfaces;
+using System;
+using APIViewWeb.Managers;
+using Microsoft.Azure.Cosmos.Serialization.HybridRow;
+
+namespace APIViewWeb.LeanControllers
+{
+    public class APIRevisionsController : BaseApiController
+    {
+        private readonly ILogger<APIRevisionsController> _logger;
+        private readonly IAPIRevisionsManager _apiRevisionsManager;
+        private readonly IReviewManager _reviewManager;
+
+
+        public APIRevisionsController(ILogger<APIRevisionsController> logger,
+            IReviewManager reviewManager,
+            IAPIRevisionsManager apiRevisionsManager)
+        {
+            _logger = logger;
+            _apiRevisionsManager = apiRevisionsManager;
+            _reviewManager = reviewManager;
+        }
+
+        /// <summary>
+        /// Endpoint used by Client SPA for listing reviews.
+        /// </summary>
+        /// <param name="pageParams"></param>
+        /// <param name="filterAndSortParams"></param>
+        /// <returns></returns>
+        [HttpPost(Name = "GetAPIRevisions")]
```

Comment:
It's a `POST` method because am passing data in the body, this is to avoid having to pass so lots of data as query parameter. The guidelines do not forbid this, neither does REST architecture. 
Am not sure what you mean by checks for `BadRequestResult` usually ASP.NET catches and return bad request automatically.

Reply:
> Am not sure what you mean by checks for BadRequestResult usually ASP.NET catches and return bad request automatically.

We should add tests to make sure this is true

Reply:
We get model validation automatically in ASP.NET API. The framework handles and returns BadRequest error so ther is no need to do this.